independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/providers/serialisers/PathPartBodyHandler.java

```diff
@@ -0,0 +1,45 @@
+package org.jboss.resteasy.reactive.common.providers.serialisers;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.nio.file.Files;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.ext.MessageBodyWriter;
+import org.jboss.resteasy.reactive.PathPart;
+
+public class PathPartBodyHandler implements MessageBodyWriter<PathPart> {
+
+    public static final int BUFFER_SIZE = 8192;
+
+    public boolean isWriteable(Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
+        return PathPart.class.isAssignableFrom(type);
+    }
+
+    public void writeTo(PathPart uploadFile, Class<?> type, Type genericType,
+            Annotation[] annotations, MediaType mediaType,
+            MultivaluedMap<String, Object> httpHeaders,
+            OutputStream entityStream) throws IOException {
+        doWrite(uploadFile, entityStream);
+    }
+
+    protected void doWrite(PathPart uploadFile, OutputStream out) throws IOException {
+        doWrite(Files.newInputStream(uploadFile.file), uploadFile.offset, uploadFile.count, out);
+    }
+
+    static void doWrite(InputStream inputStream, long offset, long count, OutputStream out) throws IOException {
+        try (InputStream in = inputStream) {
+            in.skip(offset);
+            long remaining = count;
+            byte[] buf = new byte[BUFFER_SIZE];
+            int n;
+            while ((n = in.read(buf, 0, Math.min(BUFFER_SIZE, (int) remaining))) > 0) {
```

Comment:
This needs to set a content length. You also need to guard against the count being larger than the file, if you sent a content length and then don't send the full amount of data the connection basically hangs.

Reply:
Done.